### PR TITLE
Implement named entity visibility

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Reverts changes to implement workflow visibility and implements the changes necessary to update NamedEntity visibility. For background, a NamedEntity includes a task, workflow or launch plan for a specific {Project, Domain, Name} combination across versions. NamedEntities have metadata associated with them, formerly this only included a human readable description. This change also introduces a state attribute used to control visibility in the console.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Originally these changes were implemented for Workflows. However the console uses the ListNamedEntity endpoint for resource type: workflow to list workflows in the overview page. So in light of that I had to implement the visibility changes for NamedEntities but add in validation logic to apply state changes only for workflow resource types until (if) we decide to change the visibility rules.

## Tracking Issue
https://github.com/lyft/flyte/issues/235

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/repositories/gormimpl/common.go [hunk 1]
```diff
@@ -99,12 +100,12 @@ func applyFilters(tx *gorm.DB, inlineFilters []common.InlineFilter, mapFilters [
 
 func applyScopedFilters(tx *gorm.DB, inlineFilters []common.InlineFilter, mapFilters []common.MapFilter) (*gorm.DB, error) {
 	for _, filter := range inlineFilters {
-		entityModel, ok := entityToModel[filter.GetEntity()]
+		tableName, ok := entityToTableName[filter.GetEntity()]
 		if !ok {
 			return nil, adminErrors.NewFlyteAdminErrorf(codes.InvalidArgument,
 				"unrecognized entity in filter expression: %v", filter.GetEntity())
 		}
-		tableName := tx.NewScope(entityModel).TableName()
```
Comment (katrogan) [inline pkg/repositories/gormimpl/common.go:107]: So it turns out this gorm method is buggy! I observed it always returned 'workflows' as the table name regardless of the entity model passed to it

Comment (schottra) [reply pkg/repositories/gormimpl/common.go:107]: Is there a known issue for it on the gorm GH? If not, probably should file one?

File: pkg/repositories/gormimpl/common.go [hunk 2]
```diff
@@ -99,12 +100,12 @@ func applyFilters(tx *gorm.DB, inlineFilters []common.InlineFilter, mapFilters [
...
+		println(fmt.Sprintf("Applying scoped filters for [%+v] with tableName: %+v", filter.GetEntity(), tableName))
```
Comment (schottra) [inline pkg/repositories/gormimpl/common.go:108]: Did you mean to leave this in?

File: pkg/repositories/models/named_entity.go [hunk 3]
```diff
@@ -15,6 +15,8 @@ type NamedEntityMetadataKey struct {
 // Fields to be composed into any named entity
 type NamedEntityMetadataFields struct {
 	Description string `gorm:"type:varchar(300)"`
+	// GORM doesn't save the zero value for ints, so we use a pointer for the State field
+	State *int32 `gorm:"default:0"`
```
Comment (schottra) [inline pkg/repositories/models/named_entity.go:19]: I didn't think about this before, but it seems like the type of thing we've handled before by not using `0` as a valid value for enums?

File: tests/bootstrap.go [hunk 4]
```diff
@@ -54,7 +54,7 @@ func truncateAllTablesForTestingOnly() {
 	TruncateNodeExecutionEvents := fmt.Sprintf("TRUNCATE TABLE node_execution_events;")
 	TruncateTaskExecutions := fmt.Sprintf("TRUNCATE TABLE task_executions;")
 	TruncateResources := fmt.Sprintf("TRUNCATE TABLE resources;")
-	db := database_config.OpenDbConnection(database_config.NewPostgresConfigProvider(getDbConfig(), adminScope))
+	db := database_config.OpenDbConnection(database_config.NewPostgresConfigProvider(getLocalDbConfig(), adminScope))
```
Comment (schottra) [inline tests/bootstrap.go]: Oops? 😁

File: pkg/repositories/models/named_entity.go (hunk 3 above)
Comment (katrogan) [reply pkg/repositories/models/named_entity.go:19]: this is the recommended approach for saving the default value, it's not particularly a hack to use a pointer other than it's slightly weird for golang

File: pkg/repositories/gormimpl/common.go (hunk 2 above)
Comment (katrogan) [reply pkg/repositories/gormimpl/common.go:108]: nope, thank you for the catch

File: pkg/repositories/gormimpl/common.go (hunk 1 above)
Comment (katrogan) [reply pkg/repositories/gormimpl/common.go:107]: sure

File: pkg/common/entity.go [hunk 5]
```diff
@@ -7,13 +7,14 @@ import (
...
-	TaskExecution      = "te"
-	Workflow           = "w"
+	Execution           = "e"
+	LaunchPlan          = "l"
+	NodeExecution       = "ne"
+	NodeExecutionEvent  = "nee"
+	Task                = "t"
+	TaskExecution       = "te"
+	Workflow            = "w"
+	NamedEntityMetadata = "nem"
```
[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/common/entity.go:17]: it's the convention we use for entities denoted by {Project, Domain, Name} across versions e.g. tasks, workflows, etc. Alternative suggestions are welcome, although it's a pretty engrained in flyteidl/admin/console at this point.

**Comment by katrogan** [comment]
@wild-endeavor updated the PR description
